<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1536</width>
    <height>660</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1536</width>
    <height>650</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1536</width>
    <height>660</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>ACL Editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>9</y>
      <width>1521</width>
      <height>610</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="3" column="0" colspan="3">
      <widget class="QPushButton" name="pushButton_New">
       <property name="text">
        <string>New entry</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="lineEdit_FileName">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="0" colspan="3">
      <widget class="QTableWidget" name="tableWidget_ACL">
       <column>
        <property name="text">
         <string>Trustee</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Mode</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Permissions Specific</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Permissions Standard</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Permissions Generic</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Permissions Other</string>
        </property>
       </column>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_File">
       <property name="text">
        <string>File:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="3" rowspan="4">
      <widget class="Line" name="line_11">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="pushButton_Open">
       <property name="text">
        <string>Open...</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_ACL">
       <property name="text">
        <string>DACL</string>
       </property>
      </widget>
     </item>
     <item row="3" column="6">
      <widget class="QPushButton" name="pushButton_Save">
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
     <item row="0" column="4" rowspan="3" colspan="3">
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="16" column="2" colspan="3">
        <widget class="Line" name="line_8">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="14" column="2" colspan="3">
        <widget class="Line" name="line_7">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="18" column="3">
        <widget class="QCheckBox" name="checkBox_GENERIC_EXECUTE">
         <property name="toolTip">
          <string>Execute access</string>
         </property>
         <property name="text">
          <string>GENERIC_EXECUTE</string>
         </property>
        </widget>
       </item>
       <item row="20" column="1">
        <widget class="Line" name="line_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="11" column="3">
        <widget class="QCheckBox" name="checkBox_DELETE">
         <property name="toolTip">
          <string>Delete access.</string>
         </property>
         <property name="text">
          <string>DELETE</string>
         </property>
        </widget>
       </item>
       <item row="7" column="4">
        <widget class="QCheckBox" name="checkBox_FILE_READ_ATTRIBUTES">
         <property name="toolTip">
          <string>Grants the right to read file attributes.</string>
         </property>
         <property name="text">
          <string>FILE_READ_ATTRIBUTES</string>
         </property>
        </widget>
       </item>
       <item row="22" column="2">
        <widget class="QLabel" name="label_Username">
         <property name="text">
          <string>Username</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QLabel" name="label_Specific">
         <property name="toolTip">
          <string>Specific rights. Contains the access mask specific to the object type associated with the mask (HERE - for a file or directory).</string>
         </property>
         <property name="text">
          <string>Specific</string>
         </property>
        </widget>
       </item>
       <item row="5" column="3">
        <widget class="QCheckBox" name="checkBox_FILE_WRITE_DATA">
         <property name="toolTip">
          <string>Grants the right to write data to the file.</string>
         </property>
         <property name="text">
          <string>FILE_WRITE_DATA</string>
         </property>
        </widget>
       </item>
       <item row="21" column="0" colspan="5">
        <widget class="Line" name="line_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="8" column="3">
        <widget class="QCheckBox" name="checkBox_FILE_ADD_SUBDIRECTORY">
         <property name="toolTip">
          <string>Grants the right to append data to the file. For a directory, this value grants the right to create a subdirectory.</string>
         </property>
         <property name="text">
          <string>FILE_ADD_SUBDIRECTORY</string>
         </property>
        </widget>
       </item>
       <item row="5" column="4">
        <widget class="QCheckBox" name="checkBox_FILE_TRAVERSE">
         <property name="toolTip">
          <string>Grants the right to execute a file. For a directory, the directory can be traversed.</string>
         </property>
         <property name="text">
          <string>FILE_TRAVERSE</string>
         </property>
        </widget>
       </item>
       <item row="20" column="3">
        <widget class="QComboBox" name="comboBox_AccessMode">
         <property name="toolTip">
          <string>GRANT_ACCESS		- Indicates an ACCESS_ALLOWED_ACE structure. The new ACE combines the specified rights with any existing allowed or denied rights of the trustee.
SET_ACCESS		- Indicates an ACCESS_ALLOWED_ACEstructure that allows the specified rights. On input, this value discards any existing access control information for the trustee.
DENY_ACCESS		- Indicates an ACCESS_DENIED_ACEstructure that denies the specified rights. On input, this value denies the specified rights in addition to any currently denied rights of the trustee.
REVOKE_ACCESS	- Indicates that all existing ACCESS_ALLOWED_ACE or SYSTEM_AUDIT_ACE structures for the specified trustee are removed.</string>
         </property>
         <item>
          <property name="text">
           <string/>
          </property>
         </item>
         <item>
          <property name="text">
           <string>GRANT_ACCESS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>SET_ACCESS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>DENY_ACCESS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>REVOKE_ACCESS</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="11" column="4">
        <widget class="QCheckBox" name="checkBox_WRITE_OWNER">
         <property name="toolTip">
          <string>Write access to owner.</string>
         </property>
         <property name="text">
          <string>WRITE_OWNER</string>
         </property>
        </widget>
       </item>
       <item row="6" column="4">
        <widget class="QCheckBox" name="checkBox_FILE_DELETE_CHILD">
         <property name="toolTip">
          <string>Grants the right to delete a directory and all the files it contains (its children), even if the files are read-only.</string>
         </property>
         <property name="text">
          <string>FILE_DELETE_CHILD</string>
         </property>
        </widget>
       </item>
       <item row="3" column="4">
        <widget class="QCheckBox" name="checkBox_FILE_WRITE_EA">
         <property name="toolTip">
          <string>Grants the right to write extended attributes.</string>
         </property>
         <property name="text">
          <string>FILE_WRITE_EA</string>
         </property>
        </widget>
       </item>
       <item row="23" column="2">
        <widget class="QLabel" name="label_SID">
         <property name="text">
          <string>SID</string>
         </property>
        </widget>
       </item>
       <item row="15" column="4">
        <widget class="QCheckBox" name="checkBox_MAXIMUM_ALLOWED">
         <property name="text">
          <string>MAXIMUM_ALLOWED</string>
         </property>
        </widget>
       </item>
       <item row="13" column="3">
        <widget class="QCheckBox" name="checkBox_WRITE_DAC">
         <property name="toolTip">
          <string>Write access to the DACL.</string>
         </property>
         <property name="text">
          <string>WRITE_DAC</string>
         </property>
        </widget>
       </item>
       <item row="4" column="4">
        <widget class="QCheckBox" name="checkBox_FILE_EXECUTE">
         <property name="toolTip">
          <string>Grants the right to execute a file.</string>
         </property>
         <property name="text">
          <string>FILE_EXECUTE</string>
         </property>
        </widget>
       </item>
       <item row="4" column="3">
        <widget class="QCheckBox" name="checkBox_FILE_LIST_DIRECTORY">
         <property name="toolTip">
          <string>Grants the right to read data from the file. For a directory, this value grants the right to list the contents of the directory.</string>
         </property>
         <property name="text">
          <string>FILE_LIST_DIRECTORY</string>
         </property>
        </widget>
       </item>
       <item row="9" column="3">
        <widget class="QCheckBox" name="checkBox_FILE_READ_EA">
         <property name="toolTip">
          <string>Grants the right to read extended attributes.</string>
         </property>
         <property name="text">
          <string>FILE_READ_EA</string>
         </property>
        </widget>
       </item>
       <item row="15" column="2">
        <widget class="QLabel" name="label_Other">
         <property name="text">
          <string>Other</string>
         </property>
        </widget>
       </item>
       <item row="18" column="4">
        <widget class="QCheckBox" name="checkBox_GENERIC_READ">
         <property name="toolTip">
          <string>Read access</string>
         </property>
         <property name="text">
          <string>GENERIC_READ</string>
         </property>
        </widget>
       </item>
       <item row="17" column="3">
        <widget class="QCheckBox" name="checkBox_GENERIC_ALL">
         <property name="toolTip">
          <string>All possible access rights</string>
         </property>
         <property name="text">
          <string>GENERIC_ALL</string>
         </property>
        </widget>
       </item>
       <item row="12" column="4">
        <widget class="QCheckBox" name="checkBox_SYNCHRONIZE">
         <property name="toolTip">
          <string>Synchronize access.</string>
         </property>
         <property name="text">
          <string>SYNCHRONIZE</string>
         </property>
        </widget>
       </item>
       <item row="22" column="0">
        <widget class="QLabel" name="label_Trustee">
         <property name="text">
          <string>Trustee</string>
         </property>
        </widget>
       </item>
       <item row="20" column="0">
        <widget class="QLabel" name="label_AccessMode">
         <property name="toolTip">
          <string>GRANT_ACCESS		- Indicates an ACCESS_ALLOWED_ACE structure. The new ACE combines the specified rights with any existing allowed or denied rights of the trustee.
SET_ACCESS		- Indicates an ACCESS_ALLOWED_ACEstructure that allows the specified rights. On input, this value discards any existing access control information for the trustee.
DENY_ACCESS		- Indicates an ACCESS_DENIED_ACEstructure that denies the specified rights. On input, this value denies the specified rights in addition to any currently denied rights of the trustee.
REVOKE_ACCESS	- Indicates that all existing ACCESS_ALLOWED_ACE or SYSTEM_AUDIT_ACE structures for the specified trustee are removed.</string>
         </property>
         <property name="text">
          <string>Access Mode</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1" rowspan="16">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="22" column="1" rowspan="2">
        <widget class="Line" name="line_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Owner:</string>
         </property>
        </widget>
       </item>
       <item row="17" column="2">
        <widget class="QLabel" name="label">
         <property name="toolTip">
          <string>You can use generic access rights to specify the type of access you need when you are opening a handle to an object. This is typically simpler than specifying all the corresponding standard and specific rights.</string>
         </property>
         <property name="text">
          <string>Generic</string>
         </property>
        </widget>
       </item>
       <item row="11" column="2">
        <widget class="QLabel" name="label_Standard">
         <property name="toolTip">
          <string>Standard rights. Contains the object's standard access rights.</string>
         </property>
         <property name="text">
          <string>Standard</string>
         </property>
        </widget>
       </item>
       <item row="22" column="4">
        <widget class="QPushButton" name="pushButton_CheckName">
         <property name="text">
          <string>Check name</string>
         </property>
        </widget>
       </item>
       <item row="17" column="4">
        <widget class="QCheckBox" name="checkBox_GENERIC_WRITE">
         <property name="toolTip">
          <string>Write access</string>
         </property>
         <property name="text">
          <string>GENERIC_WRITE</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_AccessPermissions">
         <property name="text">
          <string>Access Permissions</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="5">
        <widget class="Line" name="line_9">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="19" column="0" colspan="5">
        <widget class="Line" name="line_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="24" column="0" colspan="5">
        <widget class="Line" name="line_10">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="8" column="4">
        <widget class="QCheckBox" name="checkBox_FILE_WRITE_ATTRIBUTES">
         <property name="toolTip">
          <string>Grants the right to change file attributes.</string>
         </property>
         <property name="text">
          <string>FILE_WRITE_ATTRIBUTES</string>
         </property>
        </widget>
       </item>
       <item row="15" column="3">
        <widget class="QCheckBox" name="checkBox_ACCESS_SYSTEM_SECURITY">
         <property name="toolTip">
          <string>Access system security (ACCESS_SYSTEM_SECURITY). It is used to indicate access to a system access control list (SACL). This type of access requires the calling process to have the SE_SECURITY_NAME (Manage auditing and security log) privilege. If this flag is set in the access mask of an audit access ACE (successful or unsuccessful access), the SACL access will be audited.</string>
         </property>
         <property name="text">
          <string>ACCESS_SYSTEM_SECURITY</string>
         </property>
        </widget>
       </item>
       <item row="7" column="3">
        <widget class="QCheckBox" name="checkBox_FILE_APPEND_DATA">
         <property name="toolTip">
          <string>Grants the right to append data to the file. For a directory, this value grants the right to create a subdirectory.</string>
         </property>
         <property name="text">
          <string>FILE_APPEND_DATA</string>
         </property>
        </widget>
       </item>
       <item row="12" column="3">
        <widget class="QCheckBox" name="checkBox_READ_CONTROL">
         <property name="toolTip">
          <string>Read access to the owner, group, and discretionary access control list (DACL) of the security descriptor.</string>
         </property>
         <property name="text">
          <string>READ_CONTROL</string>
         </property>
        </widget>
       </item>
       <item row="22" column="3">
        <widget class="QLineEdit" name="lineEdit_Username"/>
       </item>
       <item row="23" column="3" colspan="2">
        <widget class="QLineEdit" name="lineEdit_SID">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="3">
        <widget class="QCheckBox" name="checkBox_FILE_READ_DATA">
         <property name="toolTip">
          <string>Grants the right to read data from the file.</string>
         </property>
         <property name="text">
          <string>FILE_READ_DATA</string>
         </property>
        </widget>
       </item>
       <item row="6" column="3">
        <widget class="QCheckBox" name="checkBox_FILE_ADD_FILE">
         <property name="toolTip">
          <string>Grants the right to write data to the file. For a directory, this value grants the right to create a file in the directory.</string>
         </property>
         <property name="text">
          <string>FILE_ADD_FILE</string>
         </property>
        </widget>
       </item>
       <item row="10" column="2" colspan="3">
        <widget class="Line" name="line_6">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="5">
        <widget class="Line" name="line_12">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="Line" name="line_13">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QLineEdit" name="lineEdit_Owner">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1536</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
